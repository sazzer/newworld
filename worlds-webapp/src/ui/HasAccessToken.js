// @flow

import React from 'react';
import {connectStore} from "redux-box";
import {Header, Icon, Segment} from 'semantic-ui-react';
import {Interpolate} from 'react-i18next';
import type {AuthModule} from '../auth';
import {module as authModule} from '../auth';

/**
 * Component to show if the user is not currently logged in
 */
export function MissingAccessToken() {
    return (
        <Segment raised>
            <Header as='h2' icon textAlign="center">
                <Icon name='lock' />
                <Interpolate i18nKey="unauthenticated.header" />
                <Header.Subheader>
                    <Interpolate i18nKey="unauthenticated.message" />
                </Header.Subheader>
            </Header>
        </Segment>
    );
}

/**
 * Wrapper around the a component to only render it if we've got an access token
 */
export function ConnectedHasAccessToken({auth, children}: {auth: AuthModule, children: any}) {
    return auth.selectHasAccessToken() ? children : <MissingAccessToken/>;
}

export default connectStore({auth: authModule})(ConnectedHasAccessToken);
