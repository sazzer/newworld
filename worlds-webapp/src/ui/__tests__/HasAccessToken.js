// @flow

import React from 'react';
import {shallow} from 'enzyme';
import { MissingAccessToken } from '../HasAccessToken';

/** Create the component to test */
function setup() {
    return {
        element: shallow(<MissingAccessToken />)
    };
}

describe('MissingAccessToken', () => {
    it('Renders correctly', () => {
        const { element } = setup();

        expect(element).toMatchSnapshot();
    });
});
