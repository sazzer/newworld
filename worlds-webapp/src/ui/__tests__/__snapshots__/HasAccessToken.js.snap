// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissingAccessToken Renders correctly 1`] = `
<Segment
  raised={true}
>
  <Header
    as="h2"
    icon={true}
    textAlign="center"
  >
    <Icon
      as="i"
      name="lock"
    />
    <Interpolate
      className=""
      i18nKey="unauthenticated.header"
    />
    <HeaderSubheader>
      <Interpolate
        className=""
        i18nKey="unauthenticated.message"
      />
    </HeaderSubheader>
  </Header>
</Segment>
`;
